vmm/src/vm.rs

```diff
@@ -490,6 +492,29 @@ impl UserIoapicIrq {
         UserIoapicIrq { ioapic, irq }
     }
 }
+struct WinSize {
+    ws_row: u16, // rows, in characters
+    ws_col: u16, // columns, in characters
+}
+
+impl WinSize {
+    fn new() -> WinSize {
+        WinSize {
+            ws_row: 0,
+            ws_col: 0,
+        }
+    }
+
+    fn get_win_size() -> (u16, u16) {
+        let ws: WinSize = WinSize::new();
```

Comment:
Just do let `ws: struct {rows: u16, cols: u16} = {rows: 0, cols: 0};`

Reply:
I am not sure if this is a valid syntax. But, the following works.
struct WS {rows: u16, cols: u16};
let ws: WS = WS {rows: 0u16, cols: 0u16};

Reply:
I am not sure if this is a correct syntax. The following works though.

struct WS {rows: u16, cols: u16};
let ws: WS = WS {rows: 0u16, cols: 0u16};